File: TurnierManagerBackend/src/main/java/de/jf/karlsruhe/model/enums/GameStatus.java
package de.jf.karlsruhe.model.enums;

public enum GameStatus {
    /** Das Spiel ist geplant, aber noch nicht gestartet. */
    SCHEDULED,
    /** Das Spiel läuft gerade. */
    IN_PROGRESS,
    /** Das Spiel ist beendet und das Ergebnis ist eingetragen. */
    COMPLETED,
    /** Das Spiel wurde abgesagt oder verschoben. */
    CANCELED,
    /** Das Spiel wurde beendet und mit den Spielstatistiken versehen **/
    COMPLETED_AND_STATED
}

File: frontend/lib/src/serialization/game_status.dart
enum GameStatus {
  scheduled('SCHEDULED'),
  inProgress('IN_PROGRESS'),
  completed('COMPLETED'),
  completedAndStated('COMPLETED_AND_STATED'),
  canceled('CANCELED');

  final String value;
  const GameStatus(this.value);

  static GameStatus fromString(String value) {
    return GameStatus.values.firstWhere(
      (e) => e.value == value,
      orElse: () => throw ArgumentError('Unknown GameStatus: $value'),
    );
  }
}
